src/Uno.UWP/Devices/Sensors/Gyrometer.Android.cs

```diff
@@ -0,0 +1,97 @@
+#if __ANDROID__
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Android.Hardware;
+using Android.Runtime;
+using Uno.Devices.Sensors.Helpers;
+
+namespace Windows.Devices.Sensors
+{
+	public partial class Gyrometer
+	{
+		private readonly Sensor _sensor;
+		private uint _reportInterval = 0;
+
+		private GyrometerListener _listener;
+
+		private Gyrometer(Sensor barometerSensor)
+		{
+			_sensor = barometerSensor;
+		}
+
+		public uint ReportInterval
+		{
+			get => _reportInterval;
+			set
+			{
+				lock (_syncLock)
+				{
+					_reportInterval = value;
+
+					if (_readingChanged != null)
+					{
+						//restart reading to apply interval
+						StopReading();
+						StartReading();
+					}
+				}
+			}
+		}
+
+		private static Gyrometer TryCreateInstance()
+		{
+			var sensorManager = SensorHelpers.GetSensorManager();
+			var sensor = sensorManager.GetDefaultSensor(Android.Hardware.SensorType.Gyroscope);
+			if (sensor != null)
+			{
+				return new Gyrometer(sensor);
+			}
+			return null;
+		}
+
+		private void StartReading()
+		{
+			_listener = new GyrometerListener(this);
```

Comment:
Can you add debug logging on all those methods? this will help diagnosing issues.

Reply:
Will do

Reply:
Added the logging to the common Gyrometer.cs file - OnReadingChanged and in the event add/remove